DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateProgressController.cs

```diff
@@ -34,5 +43,48 @@ public IActionResult Index(int progressId, DelegateProgressAccessRoute accessedV
             var model = new DelegateProgressViewModel(accessedVia, courseDelegatesData!);
             return View(model);
         }
+
+        [HttpGet]
+        [Route("EditSupervisor")]
+        public IActionResult EditSupervisor(int progressId, DelegateProgressAccessRoute accessedVia)
+        {
+            var centreId = User.GetCentreId();
+            var delegateCourseProgress =
+                courseService.GetDelegateCourseProgress(progressId, centreId);
+            var supervisors = userService.GetSupervisorsAtCentre(centreId);
+
+            var model = new EditSupervisorViewModel(
+                progressId,
+                accessedVia,
+                supervisors,
+                delegateCourseProgress!.DelegateCourseInfo

```

Comment:
Yes, because we are protected by the VerifyAdminUserCanAccessProgress ServiceFilter, which handles the progess == null, return NotFound case. I suppose it is technically less efficient since we're doing double database queries, but that seems to be the pattern we've decided with all these service filters.

Reply:
That makes sense, I forgot to check the controller attributes. No problem with this, then.